graphql_compiler/compiler/emit_sql.py

```diff
@@ -103,6 +112,109 @@ def _find_folded_outputs(ir):
+    entries are delimited using '|'.
+
+    All occurrences of '^', '~', and '|' in the original string values are
+    replaced with '^e' (escape), '^n' (null), and '^d' (delimiter), resp.
+
+    Undoing the encoding above as well as the XML reference entity encoding performed
+    by the XML PATH statement is deferred to post-processing when the list is retrieved
+    from the string representation produced by the subquery.
+
+    Post-processing must split on '|', convert '~' to None, and undo both the encoding above
+    and the XML reference entity encoding auto performed by XML PATH. In particular, undo
+    '^d' -> '|', '^e' -> '^', '^n' -> '~', '&amp;' -> '&', '&gt;' -> '>', '&lt;' -> '<',
+    '&0xHEX;' -> u'0xHEX'. XML post processing can be done with
+    https://docs.python.org/3/library/html.html#html.unescape
+
+    Joining from the preceding vertex to the correct output vertex
+    is performed with the WHERE clause. This is contrasted with PostgreSQL subqueries,
+    in which the join from penultimate vertex to output vertex is done within
+    the FROM/JOIN clause of the aggregation subquery.
+
+    Args:
+        output_column: SQLAlchemy Column, value being aggregated with XML PATH
+        where: SQLAlchemy BinaryExpression, predicate used to filter
+
+    Returns:
+        A SQLAlchemy Selectable corresponding to the XML PATH subquery
+    """
+    encoded_column = func.REPLACE(  # replace all occurrences of '^' in the original with '^e'
+        output_column, '^', '^e'
+    )
+    encoded_column = func.REPLACE(  # replace all occurrences of '~' in the original with '^n'
+        encoded_column, '~', '^n'
+    )
+    encoded_column = func.REPLACE(  # replace all occurrences of '|' in the original with '^d'
+        encoded_column, '|', '^d'
+    )
+
+    # delimit elements in the array using '|'and mark nulls with `~`
+    xml_column = (u'|' + func.COALESCE(  # denote null values with '~'
+        encoded_column, '~'
+    ))  # allow unambiguously distinguishing (nullable) array elements using scheme above
+
+    # use constructor: you can't directly construct an XMLPathBinaryExpression from plain text
+    xml_column = XMLPathBinaryExpression(xml_column.left, xml_column.right, xml_column.operator)
+
+    return func.COALESCE(  # coalesce to represent empty arrays as ''
+        select([xml_column]).where(where).suffix_with("FOR XML PATH ('')").as_scalar(), ''
+    )
+
+
+# 3-tuple describing the join information for each traversal in a fold.
+#
+# Contains join descriptor naming the columns used in the join predicate,
+# the source/from table, and the destination/to table
+SQLFoldTraversalDescriptor = namedtuple('SQLFoldJoinInfo', (
+    # SQLAlchemySchemaInfo join descriptor giving columns used to join from_table/to_table
+    'join_descriptor',
+
+    # Table on the left side of the join, corresponding to the outside vertex of the traversal
+    'from_table',
```

Comment:
Change this to 'from_vertex' and 'to_vertex' for consistency with the schema generation API.